File: Java/Array/SortColors.java


public class SortColors {

	public static void main(String[] args) {
		int nums[] = {2,0,1};
		sortColors(nums);
		for(int i: nums)
			System.out.println(i);
		System.out.println(Arrays.toString(nums));//This is the another method to print the array; it overrides the toString method and print the elements of the array.

	}
	public static void sortColors(int[] nums) {
        int high = nums.length-1;
        int mid = 0;
        int low = 0;
        while(mid <= high) {
            switch(nums[mid]) {
	            case 0: {
	                int temp = nums[mid];
	                nums[mid] = nums[low];
	                nums[low] = temp;
	                mid++;
	                low++;
	                break;
	            }
	            case 1: {
	                mid++;
	                break;
	            }
	            case 2: {
	                int temp = nums[mid];
	                nums[mid] = nums[high];
	                nums[high] = temp;
	                high--;
	                break;
	            }
            }
            
        }
    }

}

File: CPP/Sort Colors.cpp
\class Solution {
public:
    void sortColors(vector<int>& nums) {
        int low =0;
        int high =nums.size()-1;
        int mid = 0;
        while( mid<=high){
            switch(nums[mid]){
                //if number is 0
                case 0: 
                    swap(nums[low++],nums[mid++]);
                    break;
                 //if number is 1    
                case  1:
                    mid++;
                    break;
                 //if number is 2   
                case  2:
                    swap(nums[mid],nums[high--]);
                    break;
            }
        }
    }
};
